### ISSUE TITLE
Server no longer works v5.1.0

### ISSUE THREAD

**Issue by dshackith** [99designs/aws-vault#490]
Running on OSX Mojave, installing or upgrading the latest version of aws-vault (v5.1.0) via Homebrew:

When executing `aws-vault exec --server --debug` I get a new shell, with output indicating that server is running, but `lsof -i :9099` does not return anything, and `netstat -ltn | grep  169.254.169.254` also shows no results.
Running `aws-vault exec [profile] -- aws s3 ls` works as expected, returning the buckets that profile has access to.

Removing aws-vault and installing an earlier version via a previous version of the cask file resolves the issue and `--server` seems to work as expected.

Debug output:
```
2020/01/08 15:35:13 [keyring] Considering backends: [keychain pass file]
2020/01/08 15:35:13 Loading config file /Users/dshack/.aws/config
2020/01/08 15:35:13 Parsing config file /Users/dshack/.aws/config
2020/01/08 15:35:13 Using region "us-east-1" from AWS_REGION
2020/01/08 15:35:13 Using duration_seconds "1h0m0s" from AWS_ASSUME_ROLE_TTL
2020/01/08 15:35:13 Using a session duration of "12h0m0s" from AWS_FEDERATION_TOKEN_TTL
2020/01/08 15:35:13 Using region "us-east-1" from AWS_REGION
2020/01/08 15:35:13 Using duration_seconds "1h0m0s" from AWS_ASSUME_ROLE_TTL
2020/01/08 15:35:13 Using a session duration of "12h0m0s" from AWS_FEDERATION_TOKEN_TTL
2020/01/08 15:35:13 [keyring] Querying keychain for service="aws-vault", keychain="login.keychain"
2020/01/08 15:35:13 [keyring] Found 3 results
2020/01/08 15:35:13 Using region "us-east-1" from AWS_REGION
2020/01/08 15:35:13 Using duration_seconds "1h0m0s" from AWS_ASSUME_ROLE_TTL
2020/01/08 15:35:13 Using a session duration of "12h0m0s" from AWS_FEDERATION_TOKEN_TTL
2020/01/08 15:35:13 [keyring] Querying keychain for service="aws-vault", keychain="login.keychain"
2020/01/08 15:35:13 [keyring] Found 3 results
2020/01/08 15:35:13 profile redacted: using stored credentials
2020/01/08 15:35:13 profile redacted: using GetSessionToken (using MFA)
2020/01/08 15:35:13 profile Full: using AssumeRole (chained MFA)
2020/01/08 15:35:13 Looking for sessions for redacted
2020/01/08 15:35:13 Looking up all keys in keyring
2020/01/08 15:35:13 [keyring] Querying keychain for service="aws-vault", keychain="login.keychain"
2020/01/08 15:35:13 [keyring] Found 3 results
2020/01/08 15:35:13 Session "session,aXRoYWthc2VxdW9pYQ,YXJuOmF3czppYW06OjU5NDgxMzY5NjE5NTptZmEvZHNoYWNrZWxmb3Jk,1578533623" expires in 4h58m29.462836s
2020/01/08 15:35:13 Session Full session (1578549446) is expired, deleting
2020/01/08 15:35:13 [keyring] Removing keychain item service="aws-vault", account="Full session (1578549446)", keychain "login.keychain"
2020/01/08 15:35:13 Error deleting session: Keychain Error. (-25244)
2020/01/08 15:35:13 [keyring] Querying keychain for service="aws-vault", account="session,aXRoYWthc2VxdW9pYQ,YXJuOmF3czppYW06OjU5NDgxMzY5NjE5NTptZmEvZHNoYWNrZWxmb3Jk,1578533623", keychain="login.keychain"
2020/01/08 15:35:13 [keyring] Found item "aws-vault session for ithakasequoia"
2020/01/08 15:35:13 Re-using cached credentials ****************7LBV generated from GetSessionToken, expires in 4h58m29.423703s
2020/01/08 15:35:13 Generated credentials ****************QGGK using AssumeRole, expires in 59m59.021705s
2020/01/08 15:35:13 Local instance role server running on 127.0.0.1:9099
2020/01/08 15:35:13 Setting subprocess env: AWS_DEFAULT_REGION=us-east-1, AWS_REGION=us-east-1
```

**Comment by gws** [comment]
Seeing a similar issue on Linux on 5.1.0. Everything looks normal:

```
$ aws-vault --debug exec --server dw
2020/01/09 14:44:28 [keyring] Considering backends: [pass]
2020/01/09 14:44:28 Loading config file /home/<me>/.aws/config
2020/01/09 14:44:28 Parsing config file /home/<me>/.aws/config
2020/01/09 14:44:28 profile dw: using stored credentials
2020/01/09 14:44:28 profile dw: using GetSessionToken (using MFA)
2020/01/09 14:44:28 Looking for sessions for <account>
2020/01/09 14:44:28 Looking up all keys in keyring
2020/01/09 14:44:28 Session "[redacted]" expires in 5h35m31.373130046s
2020/01/09 14:44:29 Re-using cached credentials ****************YMJ3 generated from GetSessionToken, expires in 5h35m30.95033742s
2020/01/09 14:44:29 Starting `aws-vault server` as root in the background
2020/01/09 14:44:29 Local instance role server running on 127.0.0.1:9099
2020/01/09 14:44:29 Setting subprocess env: AWS_DEFAULT_REGION=us-west-2, AWS_REGION=us-west-2
```

... and I do get the local proxy bound to port 80.

However:

```
$ curl http://169.254.169.254/latest/meta-data/iam/security-credentials/local-credentials
Get http://127.0.0.1:9099: dial tcp 127.0.0.1:9099: connect: connection refused
```

and checking `netstat` output indicates nothing bound to port 9099.

**Comment by gws** [comment]
I'm running into a similar issue on Linux using v5.1.0.

My debug output is nearly identical to @dshackith - it indicates everything starts normally with the local instance role server running on `127.0.0.1:9099`. However, `lsof -i :9099` indicates nothing is running there.

The main difference between our situations is that my proxy host listening at `169.254.169.254:80` **is** running, but then when trying to fetch credentials:

```sh
curl http://169.254.169.254/latest/meta-data/iam/security-credentials/local-credentials
```

I get the a connection refused trying to connect to `127.0.0.1:9099`, which is understandable since nothing's listening.

Doing everything the same way on v4.7.1 works.

**Comment by mtibben** [comment]
This was caused because v5 defaults to using the exec syscall which means the aws server disappears when the exec syscall happens. Fixed in 7daf3aa7d479e48efe8e3854b01cea47e603528d
